lib/Onyx.js

```diff
@@ -681,6 +683,13 @@ function set(key, value) {
         Logger.logAlert(`Onyx.set() called after Onyx.merge() for key: ${key}. It is recommended to use set() or merge() not both.`);
     }
 
+    // If the value in the cache is the same as what we have then do not update subscribers unless they
+    // have initWithStoredValues: false then they MUST get all updates even if nothing has changed.
+    if (!cache.hasValueChanged(key, value)) {
+        notifySubscribersOnNextTick(key, value, subscriber => subscriber.initWithStoredValues === false);
```

Comment:
```suggestion
        notifySubscribersOnNextTick(key, value, !subscriber.initWithStoredValues);
```
Why does the third argument need to be a function? Can it be a simple boolean instead?

Reply:
It doesn't have to be a function. I had it as a simple boolean, but feel the function is more intuitive and also puts the context for why we are doing what we are doing where we are doing it. Check this commit to get a feel for what I mean:

https://github.com/Expensify/react-native-onyx/pull/191/commits/8c7b7d96757892ed6d982f478dc8dbe5ddc9394e